src/test/regress/regress_gp.c

```diff
@@ -628,6 +631,39 @@ repeatPalloc(PG_FUNCTION_ARGS)
 	PG_RETURN_INT32(0);
 }
 
+/*
+ * This function is a copy of resGroupPalloc that existed before the patch.
+ * It's needed for the tests, but we can't use the old name because we need
+ * to maintain compatibility with the old tests which don't expect resGroupPalloc
+ * to allocate extra memory which is startupChunks
+ */
+PG_FUNCTION_INFO_V1(resGroupPallocIgnoreStartup);
+Datum
+resGroupPallocIgnoreStartup(PG_FUNCTION_ARGS)
+{
+	float ratio = PG_GETARG_FLOAT8(0);
+	int memLimit, slotQuota, sharedQuota;
+	int size;
+	int count;
+	int i;
+
+	if (!IsResGroupEnabled())
+		PG_RETURN_INT32(0);
+
+	ResGroupGetMemInfo(&memLimit, &slotQuota, &sharedQuota);
+	size = ceilf(memLimit * ratio);
+
+	count = size / 512;
+	for (i = 0; i < count; i++)
+		MemoryContextAlloc(TopMemoryContext, 512 * 1024 * 1024);
+
+	size %= 512;
+	MemoryContextAlloc(TopMemoryContext, size * 1024 * 1024);
+
+	PG_RETURN_INT32(0);
+}
```

Comment:
Can both `resGroupPalloc...()` functions be DRYed so that they use one same function inside, maybe passing a boolean value to check if it should consider startup memory? I think it's unwanted to disperse the logic.

Reply:
No. Changing its signature would require rewriting other tests. I tried to avoid it
https://github.com/arenadata/gpdb/blob/ef84cdb85fbf05581bcbddc140c814b34487db7e/src/test/regress/regress_gp.c#L634-L639

Reply:
I meant doing something like:
```c
PG_FUNCTION_INFO_V1(resGroupPalloc);
Datum
resGroupPalloc(PG_FUNCTION_ARGS)
{
	resGroupPallocImpl(..., false);
}

PG_FUNCTION_INFO_V1(resGroupPallocIgnoreStartup);
Datum
resGroupPallocIgnoreStartup(PG_FUNCTION_ARGS)
{
	resGroupPallocImpl(..., true);
}

static ...
resGroupPallocImpl(..., bool ignore_startup)
{
	...
}
```